src/lib.rs

```diff
@@ -105,9 +107,15 @@ impl Contract {
         }
     }
 
-    pub fn set_owner(&mut self, owner_id: AccountId) {
+    pub fn set_owner(&mut self, new_owner_id: AccountId) {
         self.abort_if_not_owner();
-        self.owner_id = owner_id;
+        self.new_owner_id = new_owner_id;
+    }
+
+    pub fn accept_ownership(&mut self) {
+        assert_eq!(env::signer_account_id(), self.new_owner_id);
+        self.owner_id = self.new_owner_id.clone();
+        self.new_owner_id = "".parse().unwrap();
     }
```

Comment:
1. Let's use the following API: `propose_new_owner` + `accept_ownership`
2. Change: `signer_account_id` -> `predecessor_account_id`.
3. Rename: `new_owner_id` -> `proposed_owner_id`